vote/src/vote_account.rs

```diff
@@ -203,16 +246,26 @@ impl VoteAccounts {
         let Some(staked_nodes) = self.staked_nodes.get_mut() else {
             return;
         };
-        if let Some(node_pubkey) = vote_account.node_pubkey().copied() {
-            let Entry::Occupied(mut entry) = Arc::make_mut(staked_nodes).entry(node_pubkey) else {
-                panic!("this should not happen!");
-            };
-            match entry.get().cmp(&stake) {
+
+        VoteAccounts::do_sub_node_stake(staked_nodes, stake, vote_account.node_pubkey());
+    }
+
+    fn do_sub_node_stake(
```

Comment:
I think the 0 check should be inside here (and also in `do_add_node_stake`) since there is a precedent of not storing vote accounts with zero stake. I think this commit introduces a bug because an old stakes cache wouldn't have an entry in `staked_nodes` for a 0 staked node and this new change would expect that it exists.

Reply:
Nice catch! Tests pass so it must mean we're missing some coverage, I'll look into that too.